rfcs/2-jhipster-rfc-jhipster-companion.md

```diff
@@ -0,0 +1,142 @@
+
+## Usage in development
+
+In every JHipster app, docker-compose files are present in the `src/main/docker` folder to launch associated services such as databases, message brokers and registries using docker. Among them, a **jhipster-companion.yml** will let the users starts the **JHipster companion** which will then be available at [http://localhost:1337](http://localhost:1337). By default the docker-compose file will be generated in such a way that the companion will be able to connect to the locally running JHipster application out of the box. Another possible way to run the **JHipster companion** in development would be to develop a simple CLI that allows to start the **JHipster companion** server quickly. Such a CLI tool could be embedded into the JHipster package distributed on NPM.
+
+
+## Usage in production
+
+Once deployed on a production server, the **JHipster companion** can be configured to discover JHipster application instances automatically by plugging into service discovery backends (eureka, consul, kubernetes) or using a static list of application URLs provided in the configuration.
+
+# Reference-level explanation
+
+[reference-level-explanation]: #reference-level-explanation
+
+Implementation details will be left to the discretion of the **JHipster companion** developers but the following guidelines will need to be followed:
+
+- A separate  **JHipster companion** GitHub repository will host the application and allow collaboration
+- The **JHipster companion** application will be based on a reactive Spring Boot application using Spring Cloud Gateway as embedded proxy technology
+- The application will include Spring Cloud dependencies to allow it to plug into Eureka, Consul and Kubernetes service discovery mechanism
+- The Management UI will be developped using Angular/React/Vue.js and can reuse some the existing code that currently lives in the "Admin pages" of a JHipster application
+- The application will be packaged as a docker image and a jar file which will be published on Docker Hub and Maven Central
+- After the initial **JHipster companion** 1.0 release, it will be expected to keep backward compatibility with JHipster applications generated with JHipster version 7.0 or higher.
+- The **JHipster companion** will be an optional part of JHipster so that users will not have to use it. However, we think that it will provide sufficient improvements to the development and operational experience to convince users to adopt it.
+
+
+# Drawbacks
+
+[drawbacks]: #drawbacks
+
+- As JHipster is first and foremost a code generator, it could be argued that it is not the goal of the project to release non code generator products. Two date, several external products have been released as part of the JHipster organisation: the JHipster Registry and JHipster Console and their release cadence has been low compared to the main generator.
```

Comment:
organisation --> organization or change the organization to organisation in the **Possible future evolutions** section for consistency.